Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -8,37 +11,53 @@ import { ArtifactItemStore } from '../Store/artifactItemStore';
+            artifactEngineOptions.artifactCacheDirectory = artifactEngineOptions.artifactCacheDirectory ? artifactEngineOptions.artifactCacheDirectory : tl.getVariable("AGENT_WORKFOLDER");
             this.createPatternList(artifactEngineOptions);
+            var artifactName = sourceProvider.getRootItemPath();
+            this.artifactItemStore = new ArtifactItemStore(artifactName);
             this.artifactItemStore.flush();
             Logger.verbose = artifactEngineOptions.verbose;
             this.logger = new Logger(this.artifactItemStore);
             this.logger.logProgress();
             sourceProvider.artifactItemStore = this.artifactItemStore;
             destProvider.artifactItemStore = this.artifactItemStore;
+            if(!artifactEngineOptions.enableIncrementalDownload) {
+                artifactEngineOptions.artifactCacheDirectory = '';
+                artifactEngineOptions.artifactCacheKey = '';
+            }
+            this.cacheProvider = new CacheProvider(artifactEngineOptions.artifactCacheDirectory,artifactEngineOptions.artifactCacheKey,artifactName);
             sourceProvider.getRootItems().then((itemsToProcess: models.ArtifactItem[]) => {
                 this.artifactItemStore.addItems(itemsToProcess);
-
-                for (let i = 0; i < artifactEngineOptions.parallelProcessingLimit; ++i) {
-                    var worker = new Worker<models.ArtifactItem>(i + 1, item => this.processArtifactItem(sourceProvider, item, destProvider, artifactEngineOptions), () => this.artifactItemStore.getNextItemToProcess(), () => !this.artifactItemStore.itemsPendingProcessing());
-                    workers.push(worker.init());
-                }
-
-                Promise.all(workers).then(() => {
-                    this.logger.logSummary();
-                    sourceProvider.dispose();
-                    destProvider.dispose();
-                    resolve(this.artifactItemStore.getTickets());
-                }, (err) => {
-                    ci.publishEvent('reliability', <ci.IReliabilityData>{ issueType: 'error', errorMessage: JSON.stringify(err, Object.getOwnPropertyNames(err)) });
-                    sourceProvider.dispose();
-                    destProvider.dispose();
-                    reject(err);
+                this.createNewHashMap(sourceProvider,itemsToProcess,artifactEngineOptions).then(() => {
+                    for (let i = 0; i < artifactEngineOptions.parallelProcessingLimit; ++i) {
+                        var worker = new Worker<models.ArtifactItem>(i + 1, item => this.processArtifactItem(sourceProvider, item, destProvider, artifactEngineOptions), () => this.artifactItemStore.getNextItemToProcess(), () => !this.artifactItemStore.itemsPendingProcessing());
+                        workers.push(worker.init());
+                    }
+        
+                    Promise.all(workers).then(() => {
+                        this.logger.logSummary();
```

Comment:
logSummary published telemetry for success, we should also add if incremental download was enabled in that telemetry.